### ISSUE TITLE
ENH: Return estimate peak widths as part of find_peaks_cwt()

### ISSUE THREAD

**Issue by EfremBraun** [scipy/scipy#15060]
### Is your feature request related to a problem? Please describe.

The paper cited in the docs of https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.find_peaks_cwt.html describes how one can obtain an estimate for the peak widths. It says:
"The peak width can be estimated based on the CWT scale corresponding to the maximum point on the ridge."

This can be rather easily implemented, and I think it would be very helpful for users (including myself).

### Describe the solution you'd like.

We would just add the following to the end of find_peaks_cwt() in https://github.com/scipy/scipy/blob/master/scipy/signal/_peak_finding.py:

```
est_widths = []
for max_loc in max_locs:
    est_widths.append(widths[np.argmax(cwt_dat[:,max_loc])])
return max_locs, est_widths  
```

**Comment by EfremBraun** [comment]
@pv Looks like this might be something you're interested in.

**Comment by EfremBraun** [comment]
Actually I'm thinking that since the cited paper states:

> The Mexican Hat wavelet without scaling (a=1) provides the best matches for the peaks with the width of about two sample intervals, as shown in Figure 1. With the wavelet scale increased to a1, the peaks with bigger width, 2a1, provide the best matches.

It would be better to double the width. So the code would be what's below:

```
est_widths = []
for max_loc in max_locs:
    est_widths.append(2*widths[np.argmax(cwt_dat[:,max_loc])])
return max_locs, est_widths  
```

**Comment by NamamiShanker** [comment]
I went through the paper @EfremBraun. The ricker or Mexican Hat wavelet provides the best matches for peaks of width around 2a1. But how will we generalise it for all different kinds of wavelets?

**Comment by EfremBraun** [comment]
That's a very good point. I don't think it can be generalized for any user-defined wavelet.

I see 3 possibilities:

1. Leave the function as is, not returning widths.
2. Remove the `wavelet` parameter, forcing the user to use only the Ricker wavelet. To be honest, I very much doubt anyone uses other wavelets for this function, since none of the other SciPy wavelets given in https://docs.scipy.org/doc/scipy/reference/signal.html#wavelets are of the right form for this function. But still, it wouldn't be so nice to take away this possibility, so I don't like this option.
3. Return the estimated widths only when the default Ricker wavelet is used. This kind of feature where the type of return value depends on the input parameters is used in other SciPy functions. For example. see https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html#scipy.optimize.minimize, where the [OptimizeResult](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.OptimizeResult.html#scipy.optimize.OptimizeResult) object returned has attributes that are only available for certain choices of solver, e.g., `hess_inv`.

Perhaps option 3 would work?

**Comment by NamamiShanker** [comment]
I agree with points 2 and 3, 3rd being the better choice. What is your opinion, @tupui?

**Comment by tupui** [comment]
Sorry it's not really my area of expertise. I prefer to let other maintainers have a say. Just keep in mind backward compatibility. We cannot remove parameters without going through a deprecation cycle first. Same goes with changing the return type.
